### ISSUE TITLE
Can't use string as a HASH ref while "strict refs"

### ISSUE THREAD

**Issue by wowks** [Robertof/perl-www-telegram-botapi#16]
$VERSION = "0.07";
dies with 

Can't use string ("Inactivity timeout") as a HASH ref while "strict refs" in use at /usr/local/share/perl/5.18.2/WWW/Telegram/BotAPI.pm line 204.

Replies to die:
Inactivity timeout
Connect timeout
Bad Gateway

Is that expected or not? 
Thanks!

**Comment by Robertof** [comment]
Hi! Sorry for the delay.

This indeed looks like something that shouldn't happen! This is probably because I trust *too much* the UserAgent to give me a reference of some sort, but it's indeed not happening right now.

I intend to fix this ASAP, however due to time constraints at the moment I'm unable to provide an immediate patch.

Stay tuned and thanks for your report!

**Comment by Robertof** [comment]
Alright, here I come with an update.

On my current setup (Perl 5.22.0, Mojolicious 6.58/7.25) I currently cannot reproduce this.
It looks like the error is being caught & thrown appropriately:

```
ERROR: Inactivity timeout at WWW/Telegram/BotAPI.pm line 207.
        WWW::Telegram::BotAPI::api_request(undef, undef, HASH(0x20f4a9d0)) called at WWW/Telegram/BotAPI.pm line 76
        WWW::Telegram::BotAPI::AUTOLOAD(WWW::Telegram::BotAPI=HASH(0x20eeb790), HASH(0x20f4a9d0)) called at tg-bot-example.pl line 100
```

Would you please provide me with your Perl & Mojolicious versions? The issue seems indeed caused by [these lines](https://github.com/Robertof/perl-www-telegram-botapi/blob/a4c1b92d692d53270cdb63e42e0f94064b873f11/lib/WWW/Telegram/BotAPI.pm#L202-L205): according to your error message, it looks like the user agent is returning the error as a plain string (instead of returning false-y value and putting it in `$tx->error`). This is indeed strange. 

The strangest thing is that I remember this happening to me some time ago, but I can't reproduce it any longer.

Also, can you reproduce the issue with the following example code?

```perl
use WWW::Telegram::BotAPI;

my $api = WWW::Telegram::BotAPI->new (token => 'your_token');
$api->agent->inactivity_timeout (5);
$api->getUpdates ({ timeout => 10 });
```

Thank you!

**Comment by wowks** [comment]
Thanks for your time! I'm using clean "Ubuntu 14.04.5 LTS" install. 
As packet manager says, I have perl "5.18.2-2ubuntu1.1" and libmojolicious-perl "4.63+dfsg-1"
And yes, I can reproduce my issue with those three lines:
`
./test.pl 
Can't use string ("Inactivity timeout") as a HASH ref while "strict refs" in use at /usr/local/share/perl/5.18.2/WWW/Telegram/BotAPI.pm line 204.
`

**Comment by Robertof** [comment]
Alright, I reproduced this successfully.

The problem is caused by the *very outdated* version of Mojolicious provided by Ubuntu, which returned these kinds of errors as plain strings. This was changed in later versions, perhaps in 5.0 or something similar.

I understand that to update the module you would have to use CPAN, which is not always possible/preferable, so I am going to update the module adding a check for these conditions.

**Comment by Robertof** [comment]
I issued a commit which should solve your issue.

Could you please test the module with the most recent changes and tell me if it works for you?
Thanks!

**Comment by wowks** [comment]
Thanks a lot! 
Ancient Mojolicious version now works perfect )

`
ERROR: Inactivity timeout at /usr/local/share/perl/5.18.2/WWW/Telegram/BotAPI.pm line 208.
        WWW::Telegram::BotAPI::api_request(undef, undef, 'HASH(0x27506f8)') called at /usr/local/share/perl/5.18.2/WWW/Telegram/BotAPI.pm line 76
        WWW::Telegram::BotAPI::AUTOLOAD('WWW::Telegram::BotAPI=HASH(0x168e088)', 'HASH(0x27506f8)') called at /path/to/botname.pl line 142
`

**Comment by Robertof** [comment]
Great! I just released [v0.08](https://github.com/Robertof/perl-www-telegram-botapi/releases/tag/v0.08) which will be available on CPAN shortly.

Thanks for your efforts & for being a valuable contributor!
